snuba/query/snql/__init__.py

```diff
@@ -151,6 +162,55 @@ def visit_quoted_literal(
     ) -> Literal:
         return visit_quoted_literal(node, visited_children)
 
+    def visit_where_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Sequence[Expression], Any]
+    ) -> Sequence[Expression]:
+        _, _, conditions, _ = visited_children
+        return conditions
+
+    def visit_having_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Sequence[Expression], Any]
+    ) -> Sequence[Expression]:
+        _, _, conditions, _ = visited_children
+        return conditions
+
+    def visit_and_expression(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any, Expression]
+    ) -> Expression:
+        _, left_condition, _, and_condition = visited_children
+        # in the case of one Condition
+        # and_condition will be an empty Node
+        if not isinstance(and_condition, Node):
+            _, right_condition = and_condition
```

Comment:
Isn't `and_condition` a sequence of multiple `["AND", Expression]` ? So shouldn't you iterate over right_condition ?
If not, I am not sure I get who is combining the the multiple conditions in AND.